content/courses/native-onchain-development/serialize-instruction-data-frontend.md

```diff
@@ -517,24 +543,27 @@ const handleTransactionSubmit = async (movie: Movie) => {
         isWritable: true,
       },
       {
-        pubkey: web3.SystemProgram.programId,
+        pubkey: SystemProgram.programId,
         isSigner: false,
         isWritable: false,
       },
     ],
     data: buffer,
-    programId: new web3.PublicKey(MOVIE_REVIEW_PROGRAM_ID),
+    programId: new PublicKey(MOVIE_REVIEW_PROGRAM_ID),
   });
 
   transaction.add(instruction);
 
   try {
-    let txid = await sendTransaction(transaction, connection);
-    console.log(
-      `Transaction submitted: https://explorer.solana.com/tx/${txid}?cluster=devnet`,
+    let transactionId = await sendTransaction(transaction, connection);
+    const explorerLink = getExplorerLink(
+      "transaction",
+      transactionId,
+      "devnet",
     );
-  } catch (e) {
-    alert(JSON.stringify(e));
+    console.log(`Transaction submitted: ${explorerLink}`);
+  } catch (error) {
+    alert(JSON.stringify(error));
```

Comment:
No need for JSON stringify and per CONTRIBUTING.md you are allowed to assume anything thrown is an error.

```
catch (thrownObject) {
  const error = thrownObject as Error
  alert(error.message)
}
```

Reply:
fixed. check #346